src/appleseed/foundation/math/aabb.h

```diff
@@ -543,112 +546,175 @@ T AABB<T, N>::volume() const
     return volume;
 }
 
-template <typename T, size_t N>
-inline size_t AABB<T, N>::get_corner_count() const
+template <typename T>
+inline T half_surface_area(const AABB<T, 3>& bbox)
```

Comment:
These two methods must be moved into the next section since they're related to floating-point AABBs.